sf-stream/src/convert.rs

```diff
@@ -0,0 +1,481 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::protos::extractor;
+use aptos_api_types::{
+    Event, GenesisPayload, HashValue, MoveModuleId, ScriptPayload, Transaction, TransactionInfo,
+    TransactionPayload, WriteSet, WriteSetChange,
+};
+use std::str::FromStr;
+
+pub fn convert_move_module_id(move_module_id: &MoveModuleId) -> extractor::MoveModuleId {
+    extractor::MoveModuleId {
+        address: move_module_id.address.to_string(),
+        name: move_module_id.name.to_string(),
+        special_fields: Default::default(),
+    }
+}
+
+pub fn convert_transaction_payload(payload: &TransactionPayload) -> extractor::TransactionPayload {
+    match payload {
+        TransactionPayload::ScriptFunctionPayload(sfp) => extractor::TransactionPayload {
+            type_: protobuf::EnumOrUnknown::new(
+                extractor::transaction_payload::PayloadType::SCRIPT_FUNCTION_PAYLOAD,
+            ),
```

Comment:
This shouldn't end up as unknown correct?

Reply:
It should never- this is the field type auto generated by the protobuf generated via rust